File: ProofOfConcept/MonacoPreview/monacoPreviewHandler/monacoSRC/esm/vs/platform/instantiation/common/extensions.js
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
import { SyncDescriptor } from './descriptors.js';
const _registry = [];
export function registerSingleton(id, ctor, supportsDelayedInstantiation) {
    _registry.push([id, new SyncDescriptor(ctor, [], supportsDelayedInstantiation)]);
}
export function getSingletonServiceDescriptors() {
    return _registry;
}

File: src/modules/previewpane/FileActionsMenu/FileActionsMenu.Ui/Helpers/Extensions.cs
// Copyright (c) Microsoft Corporation
// The Microsoft Corporation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

using System;

namespace FileActionsMenu.Ui.Helpers
{
    internal static class Extensions
    {
        public static T GetOrArgumentNullException<T>(this T? value)
        {
            return value ?? throw new ArgumentNullException(nameof(value));
        }
    }
}
